lib/shopify-cli/core/monorail.rb

```diff
@@ -6,51 +6,36 @@ module ShopifyCli
   module Core
     module Monorail
       ENDPOINT_URI = URI.parse('https://monorail-edge.shopifycloud.com/v1/produce')
-      INVOCATIONS_SCHEMA = 'app_cli_command/1.0'
-      SUCCESS_SENTINEL = '_success'
-      MICROSECOND_PRECISION = 6
+      INVOCATIONS_SCHEMA = 'app_cli_command/2.0'
+
+      # Extra hash of data that will be sent in the payload
+      @metadata = {}
 
       class << self
+        attr_reader :metadata
+
         def log(name, args, &block) # rubocop:disable Lint/UnusedMethodArgument
           prompt_for_consent
-
           args = args.dup.unshift(name)
-          command_with_args = args.join(' ')
-
-          # option to add future logic to capture block result in monorail event
-          capture_result = false
-
-          start_time = Time.now.utc
+          start_time = now_in_milliseconds
           begin
-            block_result = yield
-            end_time = Time.now.utc
-            send_event(
-              schema_id: INVOCATIONS_SCHEMA,
-              payload: monorail_payload(
-                args: command_with_args,
-                duration: (end_time - start_time),
-                result: (capture_result ? block_result.to_s : SUCCESS_SENTINEL)
-              )
-            )
-            return block_result
+            result = yield
+            send_event(start_time, args)
+            return result
           rescue Exception => e # rubocop:disable Lint/RescueException
-            end_time = Time.now.utc
-            send_event(
-              schema_id: INVOCATIONS_SCHEMA,
-              payload: monorail_payload(
-                args: command_with_args,
-                duration: (end_time - start_time),
-                result: e.class.name
-              )
-            )
+            send_event(start_time, args, e.message)
             raise
```

Comment:
i know this wasn't your code but shouldn't we re-raise the same error instead of throwing a generic runtime error with raise?

Reply:
@NatMorcos `raise` will re-raise the last exception:
https://ruby-doc.org/core-2.5.1/Kernel.html#method-i-raise

Reply:
> With no arguments, raises the exception in $! or raises a RuntimeError if $! is nil

oh cool! TIL about `$!` !

Reply:
Yeah there's a whole bunch of similar globals, they come from Perl I think:

https://ruby-doc.org/docs/ruby-doc-bundle/UsersGuide/rg/globalvars.html

They're very cryptic, so the [English](https://github.com/ruby/English) library gives them useful names.